src/components/dropdown/dropdown.tsx

```diff
@@ -0,0 +1,251 @@
+import { useRef, useState, ReactNode, FC, ReactElement, KeyboardEventHandler } from 'react';
+import classNames from 'classnames/bind';
+import { Manager, Reference, Popper } from 'react-popper';
+import { useSelect } from 'downshift';
+import { Scrollbars } from 'rc-scrollbars';
+import { useOnClickOutside } from '@common/hooks';
+import { KeyCodes } from '@common/constants/keyCodes';
+import { ReactComponent as ArrowIcon } from '@assets/icons/arrow.svg';
+import { DropdownOption } from './dropdownOption';
+import { DropdownVariant, RenderDropdownOption, DropdownOptionType, DropdownValue } from './types';
+import { OPEN_DROPDOWN_KEY_CODES, CLOSE_DROPDOWN_KEY_CODES } from './constants';
+import { calculateDefaultIndex, calculateNextIndex, calculatePrevIndex } from './utils';
+import styles from './dropdown.module.scss';
+
+const cx = classNames.bind(styles);
+
+interface DropdownProps {
+  options: DropdownOptionType[];
+  value: DropdownValue;
+  disabled?: boolean;
+  error?: string;
+  mobileDisabled?: boolean;
+  title?: string;
+  touched?: boolean;
+  icon?: ReactNode;
+  variant?: DropdownVariant;
+  placeholder?: string;
+  defaultWidth?: boolean;
+  transparentBackground?: boolean;
+  className?: string;
+  toggleButtonClassName?: string;
+  onChange: (value: DropdownValue) => void;
+  onFocus?: () => void;
+  onBlur?: () => void;
+  renderOption?: RenderDropdownOption;
+}
+
+// DS link - https://www.figma.com/file/gjYQPbeyf4YsH3wZiVKoaj/%F0%9F%9B%A0-RP-DS-6?type=design&node-id=3424-12207&mode=design&t=dDq6moPaTzQLviS1-0
+// TODO: implement multiple select
+export const Dropdown: FC<DropdownProps> = ({
+  value = '',
```

Comment:
As I can see, value and options are required.
Do we really need default values?

Reply:
Actually no, but there are some time-consuming deep type errors, so I decided to temporary fix it with default value.
I'll add comment about it for further fix.